File: src/SSHServer/Packets/NewKeys.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SSHServer.Packets
{
    public class NewKeys : Packet
    {
        public override PacketType PacketType
        {
            get
            {
                return PacketType.SSH_MSG_NEWKEYS;
            }
        }

        protected override void InternalGetBytes(ByteWriter writer)
        {
            // No data, nothing to write
        }

        public override void Load(ByteReader reader)
        {
            // No data, nothing to load
        }
    }
}

File: SSHServerNodeJS/SSHServerNodeJS/Packets/NewKeys.ts
import { Packet } from "./Packet";
import { PacketType } from "./PacketType";
import { ByteReader } from "../ByteReader";
import { ByteWriter } from "../ByteWriter";

export class NewKeys extends Packet {
    public clientValue: Buffer;

    public getPacketType(): PacketType {
        return PacketType.SSH_MSG_NEWKEYS;
    }

    protected internalGetBytes(writer: ByteWriter) {
        // no data, nothing to write
    }

    public load(reader: ByteReader) {
        // no data, nothing to load
    }
}
